src/Pos/Explorer/DB.hs

```diff
@@ -133,13 +149,65 @@ instance RocksBatchOp ExplorerOp where
     toBatchOp (PutLastTxs lastTxs) =
         [Rocks.Put lastTxsPrefix (serialize' lastTxs)]
 
-    toBatchOp (UpdateAddrHistory addr txs) =
-        [Rocks.Put (addrHistoryPrefix addr) (serialize' txs)]
+    toBatchOp (UpdateAddrHistory addr txs)
+        | null txs = [Rocks.Del key]
+        | otherwise = [Rocks.Put key (serialize' txs)]
+      where
+        key = addrHistoryKey addr
 
     toBatchOp (PutAddrBalance addr coin) =
-        [Rocks.Put (addrBalancePrefix addr) (serialize' coin)]
+        [Rocks.Put (addrBalanceKey addr) (serialize' coin)]
     toBatchOp (DelAddrBalance addr) =
-        [Rocks.Del $ addrBalancePrefix addr]
+        [Rocks.Del $ addrBalanceKey addr]
+
+----------------------------------------------------------------------------
+-- Iteration
+----------------------------------------------------------------------------
+
+data BalancesIter
```

Comment:
Of course I could, `BalanceIter` is just wrong and should be called `StakesIter` (or `StakeIter`).